code/game/machinery/weathercontrol.dm

```diff
@@ -0,0 +1,184 @@
+/obj/machinery/weathercontrol
+	name = "weather control device"
+	desc = "A device which uses weather control techniques such as cloud seeding to manipulate atmospheric conditions. It runs on bluespace crystals."
+	icon = 'icons/obj/stationobjs_32x64.dmi'
+	icon_state = "wcd0"
+	use_power = 1
+	density = 1
+	anchored = 1
+	machine_flags = SCREWTOGGLE | CROWDESTROY | WRENCHMOVE | FIXED2WORK
+	req_access = list(access_rnd)
+	idle_power_usage = 100
+	var/error_message = null
+	var/efficiency_modifier = 0 //Subtract from crystal costs
+	var/crystal_reserve = 0 //Powered by bluespace crystals, this represents how much.
+
+/obj/machinery/weathercontrol/New()
+	. = ..()
+	component_parts = newlist(
+		/obj/item/weapon/circuitboard/weathercontrol,
+		/obj/item/weapon/cell/rad,
```

Comment:
Keep in mind that m*chanics will flatpack the fuck out of this for free rad cells

Reply:
This is probably not an issue since it requires you to build it first, meaning the station already has all material types and full research. Rad calls and bluespace matter bins cost trivial amounts of gold, silver, uranium (like 1/17 of a sheet). They still have to pay the plastic cost.